### DISCUSSION TITLE
Calling get_cpm.cmake multiple times

### DISCUSSION THREAD

**Discussion by itavero** [cpm-cmake/CPM.cmake#539] [General]
In our organization we are using CPM.cmake and we've ran into the situation where we pull a package with CPM.cmake, which also uses CPM.cmake to pull other packages.
Both get CPM using the `get_cpm.cmake` script.

This works without any issue, but I did notice that it starts a new download if the `CPM_DOWNLOAD_VERSION` versions do not match.
I think this can be improved, but I'm not sure what the best approach would be.

I think the `get_cpm.cmake` script should have a check that sees if CPM.cmake is already there and what version is there.
In case it's an older version than the one it is configured to download (e.g. `if(CPM_VERSION VERSION_LESS CPM_DOWNLOAD_VERSION)`), it should probably at least issue a warning.
If the versions are equal, or if the available version is newer, it should probably skip the download.

Like I said, for now it works without any issue, but it feels a bit dirty that there isn't any checks here.
I'm wondering what other's peoples thoughts on this are.

**Comment by threeal** [comment]
I think the best option to overcome this is simply to not use the `get_cpm.cmake` script at all. Instead, either include the `CPM.cmake` locally in each project or install it globally to the system.

I know it may seem ugly, but CPM is supposed to be a package manager. Think about it like pip or npm; you don't download them every time you initialize a new project, right?

**Reply by itavero** [reply]
I don't want to download it, but at the same time, I also don't want to have to think about it.
The nice thing about the script is that you just use CMake: you don't have to necessarily care or know about the fact that CPM is being used.

But I do see a potential issue if there's a different version already being loaded.

I'm not sure how other package managers handle this.
Of course, they are typically installed globally and if a definition is not compatible you'll probably get an error (and I think some have schema versions in the list of dependencies).
Given that the usage is in my opinion quite different, I'm not sure if they are trying the best reference for this.
